schema/deploy/policies/ciip_user_policies_003_restrict_permissions.sql

```diff
@@ -0,0 +1,45 @@
+-- Deploy ggircs-portal:policies/ciip_user_policies_003_restrict_permissions to pg
+-- requires: policies/ciip_user_policies_002_recreate_after_type_change
+
+/*
+Policy restricts external users from viewing data in the ciip_user table with the following rules:
+  - can see their own user data
+  - can see data for users that are approved to report for the same organisation as they are
+  - can see data for analyst users
+*/
+begin;
+
+create or replace function ggircs_portal_private.get_valid_users()
+returns setof integer as
+$fn$
+  select user_id
+  from ggircs_portal.ciip_user_organisation
+  where organisation_id
+  in (
+    select organisation_id
+    from ggircs_portal.ciip_user_organisation
+    where status = 'approved'
+    and user_id=(
+      (select id from ggircs_portal.ciip_user where uuid = ((select sub from ggircs_portal.session()))
+      )
+    )
+  )
+  union
+  select id from ggircs_portal.ciip_user where uuid ilike '%@idir';
+$fn$ language sql strict stable security definer;
+
+grant execute on function ggircs_portal_private.get_valid_users to ciip_administrator, ciip_analyst, ciip_industry_user;
+
+do
+$policy$
+declare industry_user_statement text;
+begin
+industry_user_statement := 'id in (select ggircs_portal_private.get_valid_users()) or uuid=(select sub from ggircs_portal.session())';
```

Comment:
Can/should the check for `uuid=(select sub from ggircs_portal.session())` be part of the `get_valid_users` function?  I'd expect a function called valid users to give me all the valid users including whoever's logged in

Reply:
very good point. I'll add it another union to the function to do this instead of slipping it into the using/with check statment